agent/03_agent_deployment.sh

```diff
@@ -198,15 +198,46 @@ EOF
     set -x
 }
 
+function sync_installer() {
+    if [[ ! -d "${OPENSHIFT_INSTALL_PATH}" ]]; then
+        DEST="${INSTALLER_PATH}"
+        echo "Syncing $DEST"
+
+        if [ ! -d "${DEST}" ]; then
+            mkdir -p "${DEST}"
+            git clone https://github.com/openshift/installer.git "${DEST}"
+        fi
+
+        pushd "${DEST}"
+
+        DEFAULT_BRANCH=master-agent-installer
+
+        git am --abort || true
+        git checkout "${DEFAULT_BRANCH}"
+        git fetch origin
+        git rebase "origin/${DEFAULT_BRANCH}"
+
+        # If set, use the specified PR number
+        if [ -n "${INSTALLER_PR}" ]; then
+          echo "Fetching PR ${INSTALLER_PR}"
+          git fetch origin "pull/${INSTALLER_PR}/head:pr${INSTALLER_PR}"
+          git checkout "pr${INSTALLER_PR}"
+        fi
+
+        popd
+    fi
+}
+
```

Comment:
dev-scripts already has some support for building and using the installer, I think we should try to reuse / adapt the current features as much as possible to avoid duplications. 
See [KNI_INSTALL_FROM_GIT](https://github.com/openshift-metal3/dev-scripts/blob/41b9cfb172af92788ab9c3395b2d60ae90f886f0/docs/developing-installer.md).
The [clone_installer()](https://github.com/openshift-metal3/dev-scripts/blob/41b9cfb172af92788ab9c3395b2d60ae90f886f0/ocp_install_env.sh#L77) and [build_installer()](https://github.com/openshift-metal3/dev-scripts/blob/41b9cfb172af92788ab9c3395b2d60ae90f886f0/ocp_install_env.sh#L84) functions could be reused / refactored to checkout our current development branch (instead of the master)

Reply:
That's where I copied-and-pasted the code from, but:

- `clone_installer` clones to the wrong place (path is hardcoded instead of using `$OPENSHIFT_INSTALL_PATH`) and then, worse, checks out the wrong branch (`main` or `master`). You used to be able to specify the branch but, ahem, _somebody_ [repurposed](https://github.com/openshift-metal3/dev-scripts/commit/ecf0b3807344dffe1a47e1e21d5f9d3ed9a1ba81#diff-35a54395325de0227bbdd613dbf61c3265b5f6ab205cf32348e3538399cf1b9eR276-R280) that argument :stuck_out_tongue_winking_eye:
- `build_installer` mostly does stuff we don't need. Once you take that away there's only one line left (running `hack/build.sh`), so I just inlined it.

Reply:
And that's why I suggested to refactor the code (especially for those parts that are in common for both the flows), since the requirements changed over the time - and now there's a need to use a different branch, so `sync_repo_and_patch` could be enhanced to support it, maybe through an env var given that it's not a very common request.

And currently all the functions for manipulating the installer are defined in [ocp_install_env.sh](https://github.com/openshift-metal3/dev-scripts/blob/master/ocp_install_env.sh), and in future there could be other bits that could be useful for our flows.